### PR TITLE
Support react-navigation 4.0.x (breaking change).

### PR DESCRIPTION
**Pull request by cknitt**
I assume this will also break the example. After this + all the repo/package renaming is through, we should get the example to work again.

### CODE REVIEW

**Comment by MoOx** [comment]
Should we split into several bindings (packages/repo?) Would make sense even if "annoying" to handle?

**Comment by cknitt** [comment]
For the npm package, having version numbers that match the react-navigation version should be sufficient. In the repo, we could keep a `v3` branch, and use the master branch for 4.x.

**Comment by MoOx** [comment]
I may not expressed myself correctly.

I was more thinking about react-navigation-drawer, -stack & -tabs

**Comment by cknitt** [comment]
Ah, right, sorry.

I think it's not worth the effort to have separate repos/npm packages for those, that would only complicate things further both for users and for us.

**Comment by MoOx** [comment]
I agree but people might looks for a given things & not look for it.
We could try to rename `DrawerNavigator` & friends to `ReactNavigationDrawer` maybe?

**Comment by cknitt** [comment]
Yes, that's a good idea! I'll do this in a follow-up PR.

File: package.json [hunk 1]
```diff
@@ -1,12 +1,9 @@
 {
   "name": "@reason-react-native/react-navigation",
-  "version": "3.11.0-alpha.0",
+  "version": "4.0.0-alpha.0",
   "publishConfig": {
     "access": "public"
   },
-  "peerDependencies": {
-    "react-navigation": "^3.11.0"
-  },
```
Comment (MoOx) [inline package.json]: My idea of placing peered next to the version is to ensure we follow our versioning "rule" for the binding. Any thoughts about keeping peerDep around here?

**Comment by cknitt** [comment]
@MoOx Regarding renaming: I forgot that we already have the namespace `ReactNavigation` set in `bsconfig.json`. So if we renamed `DrawerNavigator` as suggested, it would become `ReactNavigation.ReactNavigationDrawer`.

Or I would need to "namespace everything manually" instead of via `bsconfig.json` which I don't really want. So I'd rather keep things as they are.

**Comment by MoOx** [comment]
Oh yeah. I guess documentation should be ok. 
We could also go for `ReactNavigation.Drawer`.

**Pull request merged**